### PR TITLE
Documentation updates + minor fixes.

### PR DESCRIPTION
**Pull request by setrofim**

### CODE REVIEW

File: doc/source/developer_reference/writing_plugins.rst [hunk 1]
```diff
@@ -219,6 +234,32 @@ also helps WA decide how it should be handled. Currently supported kinds are:
                  processing -- this is left entirely up to the output
                  processors.
 
+As with :ref:`metrics`, artifacts are added via the context:
+
+.. code-block:: python
+
+	context.add_artifact("benchmark-output", "bech-out.txt", kind="raw",
+	                     description="stdout form running the benchmark")
```
Comment (marcbonnici) [inline doc/source/developer_reference/writing_plugins.rst]: "form" -> "from"

Comment (setrofim) [reply doc/source/developer_reference/writing_plugins.rst]: Fixed. Also added another commit with a section on metadata vs classifiers.

File: doc/source/developer_reference/writing_plugins.rst [hunk 2]
```diff
@@ -307,6 +307,54 @@ If an entry does not exit, ``update_metadata`` will create it, so it's
...
+metrics and artifacts. Classifier keys should be strings, and their values
+should be simple scalars (i.e. strings, numbers, or bools).
+
+Classifiers can be thought of as "tags" that are used to annotate metrics and
+artifacts, in order to make it easier to sort through them later. WA itself does
+not do anything with them, however output processors will augment the output
+they generate with them (for example, ``csv`` processor can add additional
+columns for classifier keys).
+
+Classifiers are typically added to by the user to attach some domain-specific
```
Comment (marcbonnici) [inline doc/source/developer_reference/writing_plugins.rst]: "added to by the user to..." <- Don't need first extra "to"

File: doc/source/developer_reference/writing_plugins.rst [hunk 3]
```diff
@@ -307,6 +307,54 @@ If an entry does not exit, ``update_metadata`` will create it, so it's
...
+:ref:`classifiers`. However, plugins can also attach additional classifiers, by
+specifying them in ``add_metric()`` and ``add_artifacts()`` calls.
+
+
+Metadata vs Classifiers
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Both metadata and classifiers are sets of essentially opaque key-value pairs
+that get included in WA output. While they may seem somewhat similar and
+interchangeable, they server different purposes and are handled differently by
```
Comment (marcbonnici) [inline doc/source/developer_reference/writing_plugins.rst]: "server" -> "serve"

**Comment by setrofim** [comment]
Comments addressed.

**Pull request merged**